File: LAB QUIZZES/CSE221_Quiz05/_A6D/_A6D_Solution.java
import java.util.*;
import java.util.function.Consumer;

public class _A6D_Solution {
    public static class Pair<K, V> {
        public K key;
        public V value;

        public Pair(K key, V value) {
            this.key = key;
            this.value = value;
        }

        public String toString() {
            return "(" + key + ", " + value + ")";
        }
    }
    public static ArrayList<Integer> solve(Integer N, int R, ArrayList<Pair<Integer, Integer>> edges) {
        ArrayList<Integer> res = new ArrayList<>();
        for (int i = 0; i <= N; i++) {
            res.add(0);
        }
        // Place your code here

        return res;
    }
}

File: LAB QUIZZES/CSE221_Quiz05/_A6D/_A6D_Solution.py
from collections import defaultdict, deque
def solve(N, R, edges):
    res = [0] * (N + 1)
    graph = defaultdict(list)
    for u, v in edges:
        graph[u].append(v)
        graph[v].append(u)
    def dfs(node, parent):
        res[node] = 1
        for n in graph[node]:
            if n != parent:
                dfs(n, node)
                res[node] += res[n]
    dfs(R, -1)
    return  res
